Client/Frontend/Browser/ClipboardBarDisplayHandler.swift

```diff
@@ -0,0 +1,82 @@
+
+import Foundation
+import Shared
+
+public struct UserDefaultClipboardKey {
+    public static let KeyLastSavedURL = "KeylastSavedURL"
+}
+
+public struct ClipboardBarToastUX {
+    static let DismissAfter = 10.0
+}
+
+protocol ClipboardBarDisplayHandlerDelegate: class {
+    func shouldDisplayClipboardBar(absoluteString: String)
+}
+
+class ClipboardBarDisplayHandler {
+    weak var delegate: ClipboardBarDisplayHandlerDelegate?
+    private var sessionStarted = true
+    private var prefs: Prefs
+    private var lastDisplayedURL: String? {
+        if let value = UserDefaults.standard.object(forKey: UserDefaultClipboardKey.KeyLastSavedURL) as? String {
```

Comment:
it's done with a UserDefaults because of the answer I had here: https://bugzilla.mozilla.org/show_bug.cgi?id=1228087#c12

`
I would think that we only show the link once. On next launch, if the link is still in the clipboard we would just ignore it since the user most likely already saw the option to navigate to it anyways.
`

Since we removed the preference I do think that displaying it every time the same URL the app is closed and then opened can get frustrating. A lot of people (including myself) have the bad habit of force closing apps instead of just tapping the home button.

But you're right about the history thing, it's a privacy problem the way it's currently implemented.

I do not have a strong opinion about any approach,  I can do it in-memory or wipe it during history cleaning. Which one do you prefer?